proposals/XXXX-unicode-normalization.md

```diff
@@ -816,6 +864,20 @@ for scalar in nfcString.unicodeScalars.normalized.nfc {
 
 `nfcString` may have an internal performance bit set so we know it is NFC. If it does, the streaming normaliser in the `for`-loop detects this and knows it doesn't need to normalise the contents a second time. It would be nice if we could generalise this, or at least take note of a normalised prefix that we can skip.
 
+### Normalizing intializers
+
+We could define initializers similar to `String(decoding:as)`, e.g. `String(normalizing:)`, which decode and transform the contents into the stdlib's preferred normal form without always needing to create an intermediary `String` instance. Such strings will compare much faster to each other, but may have different contents when viewed under the `UnicodeScalarView`, `UTF8View`, or `UTF16View`.
+
```

Comment:
My personal preference would be to see what we can do about pulling this into this pitch / proposal, or immediately afterwards, as the perf improvements to comparison, Regex matching (when we get the corresponding optimizations next release), etc. are dramatic.